rust/main/agents/scraper/src/conversions.rs

```diff
@@ -1,10 +1,13 @@
+use std::str::FromStr;
+
 use num_bigint::{BigInt, Sign};
-use sea_orm::prelude::BigDecimal;
+use sea_orm::prelude::{BigDecimal, Decimal};
 
 use hyperlane_core::U256;
 
-pub fn u256_to_decimal(v: U256) -> BigDecimal {
+pub fn u256_to_decimal(v: U256) -> Decimal {
     let mut buf = [0u8; 32];
     v.to_little_endian(&mut buf);
-    BigDecimal::from(BigInt::from_bytes_le(Sign::Plus, &buf as &[u8]))
+    let big_dec = &BigDecimal::from(BigInt::from_bytes_le(Sign::Plus, &buf as &[u8])).to_string();
+    Decimal::from_str(big_dec).unwrap()
```

Comment:
Decimal and BigDecimal are both mapped to the same type on PostgreSQL, I was wondering why the change in that too but if you look at the [definition of `Wei` (which uses BigDecimal)](https://github.com/Mantas-M/hyperlane-monorepo/blob/chore/update-seaorm/rust/main/agents/scraper/migration/src/l20230309_types.rs) that has not changed from the old version.

Reply:
Another note on this, the BigDecimal -> Decimal was a change which was made by the entity generation of the updated `sea-orm` package, and has nothing do with any specific changes made to the schema.

Reply:
I suppose that the definition of column type `Wei` did not change since it is specified in the native Postgres type: `decimal`, which is equivalent of `numeric` ("[The types decimal and numeric are equivalent](https://www.postgresql.org/docs/current/datatype-numeric.html)"). So, `Wei` defines the column type, not the type which is going to be used in Rust struct which is mapped to the record in the table.

I don't believe that Rust `Decimal` can contain all values possible in `U256`. Type `Decimal` supports only 96-bit integers. In contrast, `BigDecimal` internally contains a vector of digits (see `BigUint` type).

I wonder if it is possible to enable feature `with-bigdecimal` on Postgres driver?

Reply:
Not from what I found though going through code and docs.

Their discord is pretty dead, but it's better than nothing so I've tried to [ask there](https://discord.com/channels/873880840487206962/1315689749276790844/1315689749276790844).

In general, I doubt you would be storing gas values that overflow the decimal type in rust, perhaps one thing we could do is swap it to bigint 🤔

Reply:
Seems like we're not going to have an answer to this any time soon, so I am guessing this PR is on pause until I either burn more time trying to get it back to `BigDecimal` or we can agree on a different direction to take.

Reply:
It looks like `BigDecimal` is not supported by SeaORM at the moment. I found this issue: https://github.com/SeaQL/sea-orm/issues/1530.

We have two options to proceed:
1. Contribute code into SeaORM so that `BigDecimal` is fully supported by generation script.
2. Manually add `BigDecimal` into the generated code with adding required dependencies with an elaborated comment why we did this.

I think 2. should be fairly feasible and acceptable result.

Reply:
Nice find, I've gone ahead and implemented the second option, and ran through the tests to make sure it's working as expected.